packages/react-resizable-panels/src/scripts/persist.ts

```diff
@@ -0,0 +1,87 @@
+/**
+ * The persist function that gets inlined in server-rendered HTML.
+ * This function restores panel sizes from localStorage before React hydration.
+ *
+ * This file is the source of truth for the persist logic.
+ * It gets minified and transformed by minify-persist.ts during the build.
+ */
```

Comment:
Hey @isBatak– clarifying question about this as I'm thinking about how best to port it to #528

How does the timing for this actually work?

My understanding of server rendered React was that static HTML would be sent to the client and rendered/painted _before_ "hydration" (or any other JavaScript) actually runs. Wouldn't this injected script be too late to actually prevent the layout flicker then?